src/features/top-level-await.md

```diff
@@ -71,6 +71,18 @@ try {
 }
 ```
 
+### Polyfills
+
+You can effectively hold the execution of a static import by `await`-ing a conditional, dynamic import. This is very useful when using polyfills that you only want to load when the browser needs them:
+
+```js
+if (browserNeedsPolyfill()) {
+  await import("./polyfill.mjs");
+}
+
+import "./main.js";
```

Comment:
```suggestion
import('./main.mjs')
```

Reply:
This example works fine if you change the static import here to be a dynamic one. This would avoid the import being hoisted.

My original assessment was incorrect based on looking at this wrong. If we have no children and this is the entry point for your application this pattern works great.

We could also show if there are multiple polyfills of creating an array of polyfill promises and `await Promise.all` if we are not worried about them being in flight at the same time